syntax = "proto3";

package video_processor;

service VideoProcessor {
  rpc ProcessVideo(stream Frame) returns (stream Result) {}
}

message Frame {
  bytes image_data = 1;
  int32 width = 2;
  int32 height = 3;
  int32 channels = 4;
}

message Result {
  string data = 1;
}